stl/inc/format

```diff
@@ -366,58 +366,363 @@ private:
     ptrdiff_t _Next_arg_id = 0;
 };
 
-// TODO: test coverage
+enum class _Basic_format_arg_type : uint8_t {
+    _None,
+    _Int_type,
+    _UInt_type,
+    _Long_long_type,
+    _ULong_long_type,
+    _Bool_type,
+    _Char_type,
+    _Float_type,
+    _Double_type,
+    _Long_double_type,
+    _Pointer_type,
+    _CString_type,

```

Comment:
Suggestion: Would `_Ntcts_type` be more descriptive and less prone to confusion than `_CString_type`? "Null-Terminated Character Type Sequence" (NTCTS) is the Standard term for C-style strings.

Reply:
I'm OK with leaving it as CString in the interest of not dragging this PR out.